jdbc-interface/src/main/java/org/polypheny/db/jdbc/DbmsMeta.java

```diff
@@ -1272,6 +1272,11 @@ private void prepare( StatementHandle h, String sql ) throws NoSuchStatementExce
                 rowsChanged = num;
             }
 
+            // Some stores do not correctly report the number of changed rows (set to zero to avoid assertion error in the MetaResultSet.count() method)
+            if ( rowsChanged < 0 ) {
```

Comment:
Should we find the root cause why some stores to not report the number correctly or can we reset without furhter concerns?

Reply:
I am the one to blame for this change... PostgreSQL seems to report -1 for batch statements, and because Avatica asserts the number of rows changed to be >= 0, this is a pragmatic approach to solve at least one issue. Of course, we should also find an approach to report the correct number of affected rows.